src/gameplayv2/resistance/entities/Entity.ts

```diff
@@ -0,0 +1,37 @@
+class Entity{
+    static nextId = 0;
+    protected components: any[];
+    public role: string;
+    public alliance: string;
+    public description: string;
+
+    constructor() {
+        this.components = [];
+        this.role;
+        this.alliance;
+        this.description;
+    }
+
+    public addComponent(component: any): void {
+        this.components.push(component);
+    }
+
+    public addComponents(...components: any[]): void {
+        components.forEach((component) => this.addComponent(component));
+    }
+      
+    public removeComponent(component: any): void {
+        const componentIndex = this.components.findIndex((c) => c.constructor.name === component.constructor.name);
+        if (componentIndex !== -1) {
+            this.components.splice(componentIndex, 1);
+        }
+    }
+      
+    public getComponent(componentClass: any): any {
+        return this.components.find((c) => c.constructor.name === componentClass.name);
+    }
```

Comment:
Instead of a list, I'm starting to think it might be better to have `this.components` be a hashmap where the key is the component enum, and the value is the component itself. We can still iterate over them easily but it'll make checking for the existence of a component really fast.